### PR TITLE
Filter urllib3 InsecureRequestWarning for now

### PR DESCRIPTION
**Pull request by matthewrmshin**

### CODE REVIEW

File: lib/cylc/network/https/base_client.py [hunk 1]
```diff
@@ -22,9 +22,6 @@
 from uuid import uuid4
 import warnings
 
-# Ignore incorrect SSL certificate warning from urllib3 via requests.
-warnings.filterwarnings("ignore", "Certificate has no `subjectAltName`")
```
Comment (oliver-sanders) [inline lib/cylc/network/https/base_client.py]: Does this warning no longer require filtering?

Comment (matthewrmshin) [inline lib/cylc/network/https/base_client.py]: The `subjectAltName` field is defined in our certificate generation logic, so I don't know why we'll have this warning in the first place.

**Comment by matthewrmshin** [comment]
*urllib3* is bundled with *requests*. If we are able to import *requests*, we are able to import *urllib3* without adding extra dependency, hence the order of the statements.

**Comment by matthewrmshin** [comment]
(I can't remember why we don't do it, but if it is easy to do, we should bundle *requests* with cylc - and remove the *urllib2* code. Related to #2204 and #2250.)

File: lib/cylc/network/https/base_client.py [hunk 2]
```diff
@@ -106,6 +103,8 @@ def _get_data_from_url(self, url, json_data, method=None):
 
     def _get_data_from_url_with_requests(self, url, json_data, method=None):
         import requests
+        from urllib3.exceptions import InsecureRequestWarning
+        warnings.simplefilter("ignore", InsecureRequestWarning)
```
Comment (dvalters) [inline lib/cylc/network/https/base_client.py]: So this doesn't always work for me, I played around with it on the python interpreter and iPython and got this:
```python
In [12]: import requests     
In [13]: import urllib3.exceptions

ImportError                               Traceback (most recent call last)
<ipython-input-13-6a522130414b> in <module>()
> 1 import urllib3.exceptions

ImportError: No module named urllib3.exceptions
```
This seems more reliable though:

```python
import requests
from requests.packages.urllib3.exceptions import InsecureRequestWarning
```

Comment (matthewrmshin) [inline lib/cylc/network/https/base_client.py]: Hear hear. Statement updated.

**Pull request merged**